documentation/topics/glossary.md

```diff
@@ -0,0 +1,95 @@
+
+Actors are referenced during authorization - ensuring that the actor is allowed to perform an action, before it takes place.
+
+See the [Security guide](/documentation/topics/security.md#actors) for more information.
+
+## Aggregate
+
+An aggregate is a special type of attribute for a resource, one that summarizes related information of the record. A more specialized type of a [calculation](#calculation).
+
+If a Project resource has_many Ticket resources, an example of an aggregate on the Project might be to count the tickets associated to each project.
+
+See the [Aggregates guide](/documentation/topics/aggregates.md) for more information.
+
+## API
+
+## Attribute
+
+A piece of data belonging to a resource. The most basic building block; an attribute has a type and a value amd is stored within the context of a domain model.
+
+See `d:Ash.Resource.Dsl.attributes` for more information.
+
+## Calculation
+
+A calculation is a special type of attribute for a resource, one that is not directly stored in the data layer but generated on-demand when specifically requested as part of a query. Typically it will derive from other information on the record, but it may come from some other data source entirely. A more generic and flexible form of an [aggregate](#aggregate).
+
+See the [Calculations guide](/documentation/topics/calculations.md) for more information.
+
+## Changeset
+
+## Context
+
+## Extension
+
+## Filter
+
+## Flow
+
+See the [Flows guide](/documentation/topics/flows.md) for more information.
+
+## Identity
+
+See the [Identities guide](/documentation/topics/identities.md) for more information.
+
+## Notifier
+
+See the [Notifiers guide](/documentation/topics/notifiers.md) for more information.
+
+## Policy
+
+A set of rules defining who is authorized to perform specific actions on a resource. Common policy checks include rules such as:
+
+- Forbidding anyone other than the user who wrote a blog post, from editing it
+- Allowing only admins to update site-wide settings
+
+See the [Policies guide](/documentation/topics/policies.md) for more information.
+
+## Query
+
+## Relationship
+
+Relationships (also known as _associations_) are named links between resources, that define how they relate to each other. Relationships can be used to signify ownership of a record, membership of a group, or can be used in filtering and querying data.
+
+See the [Relationships guide](/documentation/topics/relationships.md) for more information.
+
+## Registry
+
+## Resource
+
+The central concept in Ash, a resource is a domain model object in your system, the nouns that your app revolves around. Resources contain definitions of the data they hold in the form of attributes, but also define [actions](#action) that can be taken on that data and [actors](#actor) that are allowed to run them.
```

Comment:
I was going to say this should be at the top, but then I realized this was alphabetically sorted which makes sense.

I think it would be good to hint here that resources don't necessarily need to capture state in some way, i.e a database table. There could also be a resource like `ProductEmail` that you use the `send` action on.

Reply:
That's an interesting point re: not needing state. I feel like if there's no state, then things like types of actions, identities, validations, etc. don't make sense (what are you validating? What are you CRUDing?) - which seems like a slippery slope on the way to saying "a resource can be just about anything at all". 🤔

Reply:
The action types are really just templates, they aren't meant to restrict you to working only with state that can be read back. The interface to a resource is the important part. A resource *can* be anything. Similarly, you can use an ecto schema to validate anything. For example, we're looking at CQRS/commanded integrations where a resource might have create actions but no read actions as it represents pushing to an event bus (it only tells you "success") but then other resources might have read actions but no create actions.

Reply:
So you'd be validating on arguments to actions, for example.